### DISCUSSION TITLE
no value for chat input command option

### DISCUSSION THREAD

**Discussion by ghost** [discordjs/discord-api-types#265] [Q&A] [answered]
**Please describe the problem you are having in as much detail as possible:**
hello, so, i have command, that needs to access option value, but ts yells that there is no .value for this.
**Include a reproducible code sample here, if possible:**

```ts
import { SlashCommandContext } from '../context'

export default (ctx: SlashCommandContext): Response => {
  return ctx.reply({
    content: ctx.data.options![0].value,
  })
}
```

**Further details:**
context is type which is union of `APIChatInputApplicationCommandInteraction` and two methods, reply and edit
- Runtime:
  
  - Node.js version: not node
  - deno version: not deno
- Priority this issue should have:

**Comment by vladfrangu** [comment]
Assuming you're working with our types and raw data, you'll need to cast the option as a string one or ensure it's type is string via getting the option and checking its type

**Comment by ghost** [comment]
Oh, so `data.options[0]` is string? Or what is it?

**Comment by vladfrangu** [comment, marked as answer]
No...

Either get the first option, check it's type (`if (myOption.type !== ApplicationCommandOptionTypes.String) return`) and get it's value or cast it as an ApplicationCommandInteractionDataOptionString

**Reply by ghost** [reply]
yeah worked, thanks (used the cast)
